src/engine/build.sh

```diff
@@ -270,9 +273,11 @@ case $B2_TOOLSET in
                 B2_TOOLSET_ROOT=/opt/intel/compiler50/ia32/
             fi
         fi
-        if test -r ${B2_TOOLSET_ROOT}bin/iccvars.sh ; then
+        if test -r ${B2_TOOLSET_ROOT}/setvars.sh ; then
+            . ${B2_TOOLSET_ROOT}/setvars.sh
```

Comment:
great, the last problem is that this script uses bash-ism...

oh dear, ... not sure if we want to promote the whole suite here (see line 1) to `bash` for that purpose... I will complain upstream with intel, everything else is potentially okay and can already go in.

This will already work if you apply the patch a few lines above:
```
. /opt/intel/inteloneapi/setvars.sh
./bootstrap.sh --with-toolset="intel-linux"
```

Reply:
Just after:
```
            echo "No intel compiler in current path"
            echo "Look in a few old place for legacy reason"
```
Shouldn't we change the comment then, as oneapi is not that old ?

Reply:
I think the "legacy reasons" are that the script always tried to find the paths, not that the compiler it looks for is old.